rero_ils/query.py

```diff
@@ -50,6 +50,40 @@ def inner(values):
     return inner
 
 
+def and_term_filter_fiction(field):
+    """Create a term filter for subject (fiction).
+
+    :param field: Field name.
+    :return: Function that returns a boolean AND query between term values.
+    """
+    def inner(values):
+        must = [Q('term', **{field: value}) for value in values]
+        must.append(Q(
+            "terms",
+            **{'genreForm.identifiedBy.value': current_app.config.get(
+                'FACETS_FICTION_TYPES'
+                )}))
+        return Q('bool', must=must)
+    return inner
+
+
+def and_term_filter_no_fiction(field):
+    """Create a term filter for subject (no fiction).
+
+    :param field: Field name.
+    :return: Function that returns a boolean AND query between term values.
+    """
+    def inner(values):
+        must = [Q('term', **{field: value}) for value in values]
+        for fiction_type in current_app.config.get('FACETS_FICTION_TYPES'):
+            must.append(Q('bool',
+                        must_not=Q("term", **{
+                            'genreForm.identifiedBy.value': fiction_type
+                            })))
+        return Q('bool', must=must)
+    return inner
```

Comment:
Why iterate on each term of 'FACETS_FICTION_TYPES' ?
You should use a single syntax : 
```python
must.append(Q('bool', 
        must_not=Q(
        "terms", 
            **{'genreForm.identifiedBy.value': 
            current_app.config.get('FACETS_FICTION_TYPES'
        }
    )
))
```